polynote-frontend/polynote/ui/component/state/state_handler.ts

```diff
@@ -0,0 +1,94 @@
+import * as deepEquals from 'fast-deep-equal/es6';
```

Comment:
Here's an important file - `StateHandler` objects are the way data gets pushed from the controller-y parts into the view-y parts.

Reply:
I think there's a bug in it now though, I plan to add unit tests (since it's pretty important to get the semantics right here (finally! frontend tests!)